api/app/registration/api.py

```diff
@@ -0,0 +1,296 @@
+from datetime import datetime
+import traceback
+from flask_restful import fields, marshal_with
+import flask_restful as restful
+from sqlalchemy.exc import IntegrityError
+from sqlalchemy.exc import SQLAlchemyError
+from app.utils import errors
+from app import LOGGER
+from app.registration.models import RegistrationSection
+from app.registration.models import RegistrationQuestion
+from app.registration.models import RegistrationForm
+from app.registration.models import Offer
+from app.registration.mixins import RegistrationFormMixin, RegistrationSectionMixin, RegistrationQuestionMixin
+from app.events.models import Event
+from app import db
+
+
+def registration_form_info(registration_form):
+    return {
+        'registration_form_id': registration_form.id,
+        'event_id': registration_form.event_id
+    }
+
+
+class RegistrationFormAPI(RegistrationFormMixin, restful.Resource):
+
+    registration_question_fields = {
+        'description': fields.String,
+        'type': fields.String,
+        'is_required': fields.Boolean,
+        'order': fields.Integer
+    }
+
+    registration_section_fields = {
+        'id': fields.Integer,
+        'name': fields.String,
+        'description': fields.String,
+        'order': fields.Integer,
+        'registration_questions': fields.List(fields.Nested(registration_question_fields))
+    }
+
+    registration_form_fields = {
+        'event_id': fields.Integer,
+        'registration_sections': fields.List(fields.Nested(registration_section_fields))
+    }
+
+    @marshal_with(registration_form_fields)
```

Comment:
Since we're reading the offer, this method should have the `@auth_required` decorator

Reply:
Please also add a check that the logged in user id is the same as the user_id on the offer - if not, return a 403 status